### PR TITLE
DMagic Orbital Science support.

### LINKED ISSUE #1307: Some ScienceExperimentFields suffixes don't work with Orbital Science mod
Originally reported here: https://www.reddit.com/r/Kos/comments/3vzhcw/dmagics_orbital_science/

### PR DESCRIPTION
**Pull request by tomekpiotrowski**
Fixes #1307

Adds support for Orbital Science parts, adds ScienceDataValue structure.

### CODE REVIEW

**Comment by mcortez** [comment]
Thanks for all the work.  I hadn't expected quite that much refactoring of the existing code would be required.  Orbital Science does seem fairly popular, so hopefully a lot of people will make use of all the extra work.  Thanks again!

**Comment by Dunbaratu** [comment]
@tomekpiotrowski , Do you think you can you add the use of the new PartModuleFields.ThrowIfNotCPUVessel() method that @hvacengi added recently?  (to prevent the firing off of a science experiment on different vessel).

**Comment by tomekpiotrowski** [comment]
Sure, give me a day or two

**Comment by erendrake** [comment]
@tomekpiotrowski did you want me to wait?

**Comment by tomekpiotrowski** [comment]
Ok, I've added the `ThrowIfNotCPUVessel` calls whenever a script tries to change something in a science experiment. Reading stuff will still works fine and is not limited. This is how it's supposed to work right? If so, we can merge.

**Comment by hvacengi** [comment]
I'm going to start reviewing this PR.  Depending on scope, I may ask you to merge in the develop branch after 1400 gets merged.  If it's a pretty clean merge though, I shouldn't have an issue updating it.

**Comment by Dunbaratu** [comment]
If @hvacengi is already reviewing this, should its milestone tag be changed to the v0.19.0?

**Comment by hvacengi** [comment]
Tagging as not ready until after we merge #1393 and tom is able update to be current with develop.

**Comment by tomekpiotrowski** [comment]
Updated.

File: doc/source/addons/OrbitalScience.rst [hunk 1]
```diff
@@ -0,0 +1,41 @@
+.. _orbitalscience:
+
+DMagic Orbital Science
+======================
+
+DMagic Orbital Science is a modification for Squad’s "Kerbal Space Program" (KSP) which adds extra science experiments to the game. Those experiments under the hood work differently
+than stock ones and require dedicated support (see :ref:`ScienceExperimentModule <scienceexperimentmodule>`).
+
+- Download: https://kerbalstuff.com/mod/5/DMagic%20Orbital%20Science
```
Comment (hvacengi) [inline doc/source/addons/OrbitalScience.rst]: RIP kerbalstuff :cry: I'm going to change this to Spacedock.info

Comment (tomekpiotrowski) [inline doc/source/addons/OrbitalScience.rst]: Lol, good catch :)

**Pull request merged**